### DISCUSSION TITLE
Bulk removal of records

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#5036] [Help]
Hi,

I have a large table where many of the records are of no interest to my application.

How can I remove the extra data (where _a_certain_field_ is empty)?
With a automation task?

Thank you.

**Comment by daawaan4x** [comment]
If this isn't to be done by the end user, you can check the docs for the public api for budibase.

**Comment by melohagan** [comment]
Hi @gvlx 

Unfortunately at the moment there is no API endpoint for batch deleting rows.

You could retrieve all of the rows with empty field data by doing this **POST**:

**URL** `https://budibase.app/api/public/v1/tables/{tableId}/rows/search` 
**BODY** `{
    "query": {
        "empty": {
            "a_certain_field": ""
        }
    }
}`

The _tableId_ can be found in the address bar of the builder table view. 

With this data returned, you could write a script to loop over each row, grab the *_id*, and **DELETE**:

**URL** `https://budibase.app/api/public/v1/tables/{tableId}/rows/{rowId}`

Here is the API reference for more information: https://docs.budibase.com/reference/post_tables-tableid-rows
